### PR TITLE
Rewrite from Ruby to Node.js

### PR DESCRIPTION
**Pull request by honzajavorek**
Replacing all Ruby dependencies, Aruba, and 50 lines of Ruby code by 100 lines of Node.js code and a few npm dependencies, which won't collide with any other hooks handler language.

Closes https://github.com/apiaryio/dredd-hooks-template/issues/23, closes https://github.com/apiaryio/dredd-hooks-template/issues/6

_I plan to provide Travis CI example and to update the README as a next step - noted in https://github.com/apiaryio/dredd/issues/917#issuecomment-482189943_

### CODE REVIEW

**Comment by honzajavorek** [comment]
```
security/snyk - Gemfile.lock (Apiary) — Manifest not found 
```

This is correct 😏

File: features/support/steps.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const childProcess = require('child_process');
+const { expect } = require('chai');
+const fs = require('fs-extra');
+const net = require('net');
+const which = require('which');
+const kill = require('tree-kill');
+const { Given, When, Then, Before, After } = require('cucumber');
+
+
+Before(function () {
```
Comment (kylef) [inline features/support/steps.js]: In our ES6 style guide, this and all the closures below are not permitted as they miss function names.

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,33 +7,23 @@ jobs:
...
-      - run:
-          name: Install the BDD test suite dependencies (Ruby)
-          command: bundle install
       - run:
           name: Install the reference hooks implementation (Python)
-          command: |
-            curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-            sudo python get-pip.py
-            sudo pip install dredd_hooks
+          command: sudo pip install dredd_hooks
```
Comment (kylef) [inline .circleci/config.yml]: Not saying you have to change this, but you can avoid using sudo with:

```shell
$ pip install dredd_hooks --user
```

Then to use any binary it provides:

```shell
$ export PATH="$HOME/.local/bin:$PATH"
```

I believe Python would automatically pick up the library from `~/.local`

Comment (honzajavorek) [reply .circleci/config.yml]: That's neat. I think it doesn't matter in CI, but it's good to know this is so easy now. (Or was it the same in the past as well? 🤔 )

File: package.json [hunk 3]
```diff
@@ -2,14 +2,22 @@
   "name": "dredd-hooks-template",
   "version": "1.0.0",
   "description": "A meta-package to install the testing dependencies of the Dredd Hooks template test suite",
+  "engines": {
+    "node": ">=10"
```
[the remark this answers is not in the document]
Comment (honzajavorek) [reply package.json]: Good point, addressed.

**Comment by honzajavorek** [comment]
@kylef I plan to add ESlint, but it's one of the last points in https://github.com/apiaryio/dredd/issues/917#issuecomment-482189943. I didn't want to get hold back by that now. Thanks for the review!

**Pull request merged**